### ISSUE TITLE
Voice node is incompatible with Conversation node

### ISSUE THREAD

**Issue by arneman** [guidone/node-red-contrib-chatbot#330]
When using Conversation node i get a msg like this:
```
{ ...
  originalMessage: { chatId: "151561651",
                                userId: null,
                                messageId: null,
                                transport: "telegram",
                                language: null }
```

**Issue:**
The voice node tries to extract chatId from `msg.payload.chatId`, or (if null) in `originalMessage.chat.id` (not `originalMessage.chatId`).

`var chatId = msg.payload.chatId || (originalMessage && originalMessage.chat.id);`

`messageId` is also wrong:
`var messageId = msg.payload.messageId || (originalMessage && originalMessage.message_id);`

And another issue: If you use payload.chatId, you are not able to set the text anymore, because it will only look in msg.payload for the text.

I would suggest:
```
      var chatId = msg.payload.chatId || (originalMessage && originalMessage.chatId);
      var messageId = msg.payload.messageId || (originalMessage && originalMessage.messageId);
      var template = MessageTemplate(msg, node);

      // check transport compatibility
      if (!utils.matchTransport(node, msg)) {
        return;
      }

      if (!_.isEmpty(node.message)) {
        message = node.message;
      } else if (_.isString(msg.payload) && !_.isEmpty(msg.payload)) {
        message = msg.payload;
      } else if (_.isString(msg.payload.text) && !_.isEmpty(msg.payload.text)) {
        message = msg.payload.text;
      } else {
        node.error('Empty message');
      }
```

**Comment by arneman** [comment]
btw: my workaround at the moment is to use a function node in before of the voice node:
```
msg.payload =  msg.msg.text; //that is where the text is
msg.originalMessage.chat = {id: msg.originalMessage.chatId};
```

**Comment by guidone** [comment]
Hi @arneman I'll look into this soon, thanks for the feedback

**Comment by guidone** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Update Voice node](https://trello.com/c/d6EG6orK/5-update-voice-node)

**Comment by guidone** [comment]
Fixed in `0.17.4`, thanks for the feedback
